### PR TITLE
fix(deployment): bonita portal as remote client of the bonita engine

### PR DESCRIPTION
**Pull request by tbouffard**
The previous way of document this was mixed with how to configure custom
code being client of the Bonita Engine.
In addition, provide explanation for both the Tomcat and Wildfly bundles

### CODE REVIEW

**Comment by tbouffard** [comment]
**Important**: modifications must be applied to the 7.9 backport in the Tomcat bundle Bonita Portal configuration as we have switched from Bitronix to Arjuna/Narayana
The introduction must also be changed as there is no more `deploy` bundle

File: md/configure-client-of-bonita-bpm-engine.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ If you want your application to use the engine API of a remote engine, your appl
 
 The engine (the *server*) must also be configured to accept the connection.
 
+**Note**: the Bonita Portal can be configured to be used as a Client of the Bonita Engine. See the [Alternate Bonita Deployment](two-main-types-of-deployment.md)
```
Comment (ecuriotto) [inline md/configure-client-of-bonita-bpm-engine.md:7]: Is it just the Bonita Portal or all the web layer?

File: md/two-main-types-of-deployment.md [hunk 2]
```diff
@@ -50,7 +45,150 @@ But you can still use the pre packaged Tomcat bundles or the `bonita.war` webapp
...
+```
+
+On Windows (start-bonita.bat)
+```
+   call setup\setup.bat init %0 %1 %2 %3 %4 %5 %6 %7 %8 %9
+   if errorlevel 1 (
+       goto exit
+   )
+```
+
```
Comment (ecuriotto) [inline md/two-main-types-of-deployment.md]: Should we add in this paragraph to remove the setup folder?

File: md/two-main-types-of-deployment.md [hunk 3]
```diff
@@ -50,7 +45,150 @@ But you can still use the pre packaged Tomcat bundles or the `bonita.war` webapp
...
+        </datasources>
+    </subsystem>
+```
+
+
+Edit the `<bonita-installation-directory>/server/standalone/deployments/bonita-all-in-one-x.y.z.ear/bonita.war/WEB-INF/web.xml`
+
+Apply the same changes as for the Tomcat bundle to disable Engine resources
+
+
```
Comment (ecuriotto) [inline md/two-main-types-of-deployment.md:183]: In the "Configuring the Bonita Stack" chapter there's no mention to changes in the server with the engine. Is it correct?

File: md/two-main-types-of-deployment.md [hunk 4]
```diff
@@ -26,20 +26,15 @@ It is fast because the Bonita Portal and the Bonita Engine run on the same JVM a
...
-
-::: info
-Don't forget to set in this JVM the following properties:
- * `org.bonitasoft.platform.username` must be set with the username of the platform admin
- * `org.bonitasoft.platform.password` must be set with the password of the platform admin
-:::
-
 ### Accessible through HTTP
 
 ![deploy2](images/images-6_0/poss_deploy2.png)
```
Comment (ecuriotto) [inline md/two-main-types-of-deployment.md:38]: It should be good to add to the picture a reference to the two application servers and then use it in the chapter. If we use "Bonita portal", this term should appear in the picture

**Review by ecuriotto — COMMENTED** [review]
Very good, following these instructions I could setup a working environment with client and engine in separate servers

File: md/configure-client-of-bonita-bpm-engine.md (hunk 1 above)
Comment (tbouffard) [reply md/configure-client-of-bonita-bpm-engine.md:7]: In the documentation, we only talk about the Portal.

File: md/two-main-types-of-deployment.md (hunk 4 above)
Comment (tbouffard) [reply md/two-main-types-of-deployment.md:38]: I don't understand `portal` is already present in all pictures of this page

image

File: md/two-main-types-of-deployment.md (hunk 3 above)
Comment (tbouffard) [reply md/two-main-types-of-deployment.md:183]: Yes, that's correct
I will add a section about this to make this explicit

File: md/two-main-types-of-deployment.md (hunk 2 above)
Comment (tbouffard) [reply md/two-main-types-of-deployment.md]: I document to use the setup tool to configure the Bonita Portal bundle but for now this is only for the setenv file
I have to check how we document the alternative authentication configuration which is a mix between server side and Bonita configuration (which requires configuration in the setup tool only used with the Engine part here)

Comment (tbouffard) [reply md/two-main-types-of-deployment.md]: done with an additional commit

**Comment by tbouffard** [comment]
Merging to propose the solution that has been validated by @ecuriotto 
Feel free to open new pull request to improve this documentation

**Pull request merged**